pkg/img/client/client.go

```diff
@@ -0,0 +1,58 @@
+package client
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/containerd/containerd/snapshots/overlay"
+	"github.com/moby/buildkit/control"
+	"github.com/moby/buildkit/session"
+	"github.com/moby/buildkit/worker/base"
+	"github.com/sirupsen/logrus"
+
+	"github.com/dominodatalab/forge/pkg/img/types"
+)
+
+// Client holds the information for the client we will use for communicating
+// with the buildkit controller.
+type Client struct {
+	backend string
+	root    string
+
+	sessionManager *session.Manager
+	controller     *control.Controller
+	workerOpt      *base.WorkerOpt
+}
+
+// New returns a new client for communicating with the buildkit controller.
+func New(root, backend string) (*Client, error) {
+	// Set the name for the directory executor.
+	name := "runc"
+
+	switch backend {
+	case types.AutoBackend:
+		if overlay.Supported(root) == nil {
+			backend = types.OverlayFSBackend
+		} else {
+			backend = types.NativeBackend
+		}
```

Comment:
From perusing the docs it sounds like native is super slow, do we need a big red warning of some type if it's enabled?

Reply:
Does a "big red warning" meaning a log statement?

Reply:
At the moment, yeah, might be reevaluating if we even want / need to support it.